### PR TITLE
fix: Fixed the problem that internationalization of table columns fails when searching

### LINKED ISSUE #48110: 表格列在搜索情况下，国际化失效
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/shu-xing-shai-xuan-cai-dan-antd-5-15-3-forked-krxdlp?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu96vpho0006356g14u7pcga%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu96vpho0002356g58lnhe9d%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu96vpho0003356gfe0ny54q%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu971a7700er356gttdgg7e0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu96vpho0002356g58lnhe9d%2522%253A%257B%2522id%2522%253A%2522clu96vpho0002356g58lnhe9d%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu96ws3k0002356g52rnd7wx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A18%252C%2522startColumn%2522%253A8%252C%2522endLineNumber%2522%253A18%252C%2522endColumn%2522%253A8%257D%255D%252C%2522filepath%2522%253A%2522%252Fdemo.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clu96ws3k0002356g52rnd7wx%2522%257D%252C%2522clu96vpho0003356gfe0ny54q%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clu96vpho0003356gfe0ny54q%2522%257D%252C%2522clu971a7700er356gttdgg7e0%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu9714hj00en356glf21u1pi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clu971a7700er356gttdgg7e0%2522%252C%2522activeTabId%2522%253A%2522clu9714hj00en356glf21u1pi%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

1. 展开搜索项 2. 搜索111 3. 显示NOT FOUND

### What is expected?

国际化显示正确

### What is actually happening?

国际化显示错误

| Environment | Info |
| --- | --- |
| antd | 5.13.0 |
| React | 18.2.0 |
| System | windows |
| Browser | 123.0.6312.86（正式版本） （64 位） |

### PR DESCRIPTION
**Pull request by LingJinT**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close [#48110](https://github.com/ant-design/ant-design/issues/48110)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fixed the problem that internationalization of table columns fails when searching      |
| 🇨🇳 Chinese |     修复表格列在搜索情况下，国际化失效      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/table/hooks/useFilter/FilterDropdown.tsx [hunk 1]
```diff
@@ -444,26 +455,23 @@ function FilterDropdown<RecordType>(props: FilterDropdownProps<RecordType>) {
...
+              onDeselect={onSelectKeys}
+              selectedKeys={selectedKeys}
+              getPopupContainer={getPopupContainer}
+              openKeys={openKeys}
+              onOpenChange={onOpenChange}
+              items={items}
+            />
+          ) : (
+            empty
+          )}
```
Comment (afc163) [inline components/table/hooks/useFilter/FilterDropdown.tsx:474]: 不太对劲，原来的 not found 是哪来的呢？

**Comment by li-jia-nan** [comment]
pr title 需要用英文

File: components/table/hooks/useFilter/FilterDropdown.tsx (hunk 1 above)
Comment (afc163) [inline components/table/hooks/useFilter/FilterDropdown.tsx:474]:
```suggestion
          {isEmpty ? empty : (
            <Menu
              selectable
              multiple={filterMultiple}
              prefixCls={`${dropdownPrefixCls}-menu`}
              className={dropdownMenuClass}
              onSelect={onSelectKeys}
              onDeselect={onSelectKeys}
              selectedKeys={selectedKeys}
              getPopupContainer={getPopupContainer}
              openKeys={openKeys}
              onOpenChange={onOpenChange}
              items={items}
            />
          )}
```

**Comment by afc163** [comment]
图片

需要测试用例。

File: components/table/hooks/useFilter/FilterDropdown.tsx (hunk 1 above)
Comment (LingJinT) [reply components/table/hooks/useFilter/FilterDropdown.tsx:474]: css伪元素写的，content为not found

Comment (afc163) [reply components/table/hooks/useFilter/FilterDropdown.tsx:474]: 确实。

Comment (LingJinT) [reply components/table/hooks/useFilter/FilterDropdown.tsx:474]: 这么改可以吗？有没有更好的方案？

Comment (afc163) [reply components/table/hooks/useFilter/FilterDropdown.tsx:472]: 可行的。

Comment (LingJinT) [reply components/table/hooks/useFilter/FilterDropdown.tsx:472]: done~

**Review by afc163 — APPROVED** [review]

File: components/table/__tests__/Table.filter.test.tsx [hunk 2]
```diff
@@ -2079,6 +2079,44 @@ describe('Table.filter', () => {
       fireEvent.change(container.querySelector('.ant-input')!, { target: { value: '111' } });
     });
 
+    it('renders empty element when search not found', () => {
+      jest.spyOn(console, 'error').mockImplementation(() => undefined);
```
Comment (crazyair) [inline components/table/__tests__/Table.filter.test.tsx:2083]: 这个不是 mockRestore 清理掉吗

Comment (afc163) [reply components/table/__tests__/Table.filter.test.tsx:2083]: 得清理

Comment (li-jia-nan) [reply components/table/__tests__/Table.filter.test.tsx:2083]: @LingJinT 来补一个？

**Pull request merged**